### ISSUE TITLE
Delegated Dependencies include dependencies for all platforms.

### ISSUE THREAD

**Issue by ThadHouse** [wpilibsuite/GradleRIO#151]
Working on adding desktop build, I noticed MSVC builds were failing. Looking at the logs, Desktop libraries were being linked into the RoboRIO build (this was not failing), and Rio libraries were being linked into the desktop build (this was failing). 

All that was done was adding `targetPlatform 'desktop'` to a C++ build with the newest GradleRIO.

**Comment by ThadHouse** [comment]
Note its not specific to the JSON stuff I'm working on, as its adding wpilib deps as well.

```
-o
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\exe\\frcUserProgram\\roborio\\frcUserProgram
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\objs\\frcUserProgram\\roborio\\frcUserProgramCpp\\biznmyfp3z431mmtf5jlwqjsh\\main.o
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\wpilibc-2018.4.1-windowsx86-64.zip_4046760d02bba87cf419ea8487107d64\\windows\\x86-64\\shared\\wpilibc.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\wpilibc-2018.4.1-windowsx86-64.zip_4046760d02bba87cf419ea8487107d64\\windows\\x86-64\\static\\wpilibc.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\hal-2018.4.1-windowsx86-64.zip_e9cc46a9246ea24265c6e8521bdc15d4\\windows\\x86-64\\shared\\wpiHal.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\hal-2018.4.1-windowsx86-64.zip_e9cc46a9246ea24265c6e8521bdc15d4\\windows\\x86-64\\static\\wpiHal.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\wpiutil-cpp-3.2.0-windowsx86-64.zip_fce6ddee37aa2209a29bc397e1cdbd50\\windows\\x86-64\\shared\\wpiutil.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\wpiutil-cpp-3.2.0-windowsx86-64.zip_fce6ddee37aa2209a29bc397e1cdbd50\\windows\\x86-64\\static\\wpiutil.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ntcore-cpp-4.1.0-windowsx86-64.zip_4ba95f083c866153120a7c54a35bf646\\windows\\x86-64\\shared\\ntcore.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ntcore-cpp-4.1.0-windowsx86-64.zip_4ba95f083c866153120a7c54a35bf646\\windows\\x86-64\\static\\ntcore.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\cscore-cpp-1.3.0-windowsx86-64.zip_90a26632bc65f996d0ade5630f1b7323\\windows\\x86-64\\shared\\cscore.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\cscore-cpp-1.3.0-windowsx86-64.zip_90a26632bc65f996d0ade5630f1b7323\\windows\\x86-64\\static\\cscore.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_calib3d320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_core320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_features2d320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_flann320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_highgui320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_imgcodecs320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_imgproc320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_ml320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_objdetect320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_photo320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_shape320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_stitching320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_superres320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_video320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_videoio320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\shared\\opencv_videostab320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-windowsx86-64.zip_6c914e7ab7824a12b5a05c05fb786160\\windows\\x86-64\\static\\opencv320.lib
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\wpilibc-2018.4.1-linuxathena.zip_271aadd5fe68d534a7ec7d8bcf00a5f6\\linux\\athena\\shared\\libwpilibc.so
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\hal-2018.4.1-linuxathena.zip_ec5b4e6116b39724b13940cebe9e01a1\\linux\\athena\\shared\\libwpiHal.so
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\wpiutil-cpp-3.2.0-linuxathena.zip_b3c0854cbf80984db4dae519ac88ea3b\\linux\\athena\\shared\\libwpiutil.so
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ntcore-cpp-4.1.0-linuxathena.zip_25c0a1cfddd66d5210163e2a58b2dd47\\linux\\athena\\shared\\libntcore.so
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\cscore-cpp-1.3.0-linuxathena.zip_5d3f92c07ea1bd9fe3d09c0df5c59501\\linux\\athena\\shared\\libcscore.so
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_calib3d.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_core.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_features2d.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_flann.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_highgui.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_imgcodecs.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_imgproc.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_java320.so
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_ml.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_objdetect.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_photo.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_shape.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_stitching.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_superres.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_video.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_videoio.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\opencv-cpp-3.2.0-linuxathena.zip_882ce6d6786024fd1378ddee15c75ec3\\linux\\athena\\shared\\libopencv_videostab.so.3.2
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ni-libraries-2018.4.1-linuxathena.zip_2c64ecd4b3048d6e908a9a8357e6be0a\\linux\\athena\\shared\\libFRC_NetworkCommunication.so.18.0.0
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ni-libraries-2018.4.1-linuxathena.zip_2c64ecd4b3048d6e908a9a8357e6be0a\\linux\\athena\\shared\\libNiFpga.so.17.0.0
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ni-libraries-2018.4.1-linuxathena.zip_2c64ecd4b3048d6e908a9a8357e6be0a\\linux\\athena\\shared\\libNiFpgaLv.so.17.0.0
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ni-libraries-2018.4.1-linuxathena.zip_2c64ecd4b3048d6e908a9a8357e6be0a\\linux\\athena\\shared\\libNiRioSrv.so.17.0.0
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ni-libraries-2018.4.1-linuxathena.zip_2c64ecd4b3048d6e908a9a8357e6be0a\\linux\\athena\\shared\\libRoboRIO_FRC_ChipObject.so.18.0.0
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ni-libraries-2018.4.1-linuxathena.zip_2c64ecd4b3048d6e908a9a8357e6be0a\\linux\\athena\\shared\\libniriodevenum.so.17.0.0
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ni-libraries-2018.4.1-linuxathena.zip_2c64ecd4b3048d6e908a9a8357e6be0a\\linux\\athena\\shared\\libniriosession.so.17.0.0
C:\\Users\\thadh\\Documents\\JavaDebugTest\\build\\tmp\\expandedArchives\\ni-libraries-2018.4.1-linuxathena.zip_2c64ecd4b3048d6e908a9a8357e6be0a\\linux\\athena\\shared\\libvisa.so
--sysroot
C:\\frc
-pthread
-rdynamic

```
Heres a log.

**Comment by JacisNonsense** [comment]
Fixed in ET 2018.09.27 (commit 821409a)
